crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -251,17 +255,19 @@ where
         block_gas_capacity: u64,
         block_bytes: u64,
         block_fees: u64,
+        gas_price: u64,
     ) -> anyhow::Result<()> {
         let capacity = Self::validate_block_gas_capacity(block_gas_capacity)?;
         let mut storage_tx = self.storage_tx_provider.begin_transaction()?;
-        let mut new_recorded_height = match storage_tx
+        let (old_recorded_height, mut new_recorded_height) = match storage_tx
             .get_recorded_height()
             .map_err(|err| anyhow!(err))?
         {
-            Some(_) => None,
+            Some(old) => (Some(old), None),
             None => {
                 // Sets it on first run
-                self.initial_recorded_height.take()
+                let initial = self.initial_recorded_height.take();
+                (initial, initial)
             }
         };
```

Comment:
do we need this change?

Reply:
There might be a better way of expressing this. It's to give the value here:
```rs
        let best_recorded_height = new_recorded_height.or(old_recorded_height);
```